0004-post-tags.md

```diff
@@ -0,0 +1,324 @@
+
+To fill all needed use cases 3 tag flavors will be needed:
+
+- NSFW
+- Content Warning
+- Generic Tag
+
+Theoretically NSFW could be implemented using a preset "Content Warning" tag but seperating out this tag allows instances to better filter it out for moderation purposes (for example if no admin/moderator is willing to moderate NSFW content).
+Both "NSFW" and "Content Warning" tags should blur the post body by default. Additionally the `sensitive` post flag to `true` should either of these flavors be present in the post tags to ensure correct handling on other fediverse platforms.
+
+Example Tag Objects:
+
+**Generic Tag:**
+```json
+{
+    "id": 1,
+    "url": "https://example.org/c/community/t/tag",
+    "name": "Generic Tag"
+}
+```
+
+**NSFW Tag:**
+```json
+{
+    "flavor": "nsfw",
+    "id": 1,
+    "url": "https://example.org/c/adult_community/t/nsfw",
+    "name": "NSFW"
+}
+```
+
+**Spoiler Tag:**
+```json
+{
+    "flavor": "cw",
+    "id": 1,
+    "url": "https://example.org/c/story_community/t/spoiler",
+    "name": "Spoiler"
+}
+```
+
+**Remote Instance Spoiler Tag:**
+```json
+{
+    "flavor": "cw",
+    "id": 1,
+    "url": "https://example.org/c/example_community/tag@example_remote.org",
+    "name": "Spoiler"
+}
+```
+
+Below is an example of a News post in the news community. The post has a communtiy tag for the newspaper company that published it, as well as a content warning for Blood/Gore. In order to share this news post across instances it also is tagged with the news tag of a remote instance.
+
+Example Post json:
+
+``` json
+{
+  "content": ". . .",
+  "sensitive": true,
+  "tag": [
+    {
+      "id": 1,
+      "url": "https://example.org/c/news/t/newspaper_company_1",
+      "name": "Newspaper Company Tag"
+    },
+    {
+      "flavor": "cw",
+      "id": 2,
+      "url": "https://example.org/c/viral_clips/t/blood",
+      "name": "Blood/Gore"
+    },
+    {
+      "id": 3,
+      "url": "https://example.org/c/tv_shows/t/news@example_remote.org",
+      "name": "News"
+    }
+  ]
+}
+```
+
+### Backend:
+I am not quite familiar with the Lemmy Backend so any corrections or additions for this section are appreciated.
+
+One new tables for "community_tags" would be required. This new table would consist of the columns `url`, `name`, `flavor`, `deleted`, `ìd` and `community_id`. Additional columns for tag display style (for example background & text color) could be added later by extending this table.
```

Comment:
Clarify these are tags for a post within a community:

````suggestion
One new table for "community_post_tag" would be required:

```sql
CREATE TABLE community_post_tag (
  id bigserial PRIMARY KEY,
  community_id bigint REFERENCES community.id,
  ap_id text UNIQUE NOT NULL,
  name text NOT NULL,
  published timestamptz NOT NULL,
  deleted timestamptz NULL,
);
```
  

Additional columns for tag display style (for example background & text color) and hierarchy could be added later by extending this table.
````